src/test/java/org/isf/opd/test/Tests.java

```diff
@@ -261,6 +334,465 @@ public void testListenerShouldUpdatePatientToMergedWhenPatientMergedEventArrive(
 		assertThat(result.getPatient().getCode()).isEqualTo(mergedPatient.getCode());
 	}
 
+	@Test
+	public void testMgrGetOpd() throws Exception {
+		int code = _setupTestOpd(false);
+		Opd foundOpd = opdIoOperationRepository.findOne(code);
+		ArrayList<Opd> opds = opdBrowserManager.getOpd(
+				foundOpd.getDisease().getType().getCode(),
+				foundOpd.getDisease().getCode(),
+				foundOpd.getVisitDate(),
+				foundOpd.getVisitDate(),
+				foundOpd.getAge() - 1,
+				foundOpd.getAge() + 1,
+				foundOpd.getSex(),
+				foundOpd.getNewPatient());
+		assertThat(opds.get(opds.size() - 1).getCode()).isEqualTo(foundOpd.getCode());
+	}
+
+	@Test
+	public void testMgrGetOpdListPatientId() throws Exception {
+		int code = _setupTestOpd(false);
+		Opd foundOpd = opdIoOperationRepository.findOne(code);
+		ArrayList<Opd> opds = opdBrowserManager.getOpdList(foundOpd.getPatient().getCode());
+		assertThat(opds.get(opds.size() - 1).getCode()).isEqualTo(foundOpd.getCode());
+	}
+
+	@Test
+	public void testMgrGetOpdListPatientIdZero() throws Exception {
+		Patient patient = testPatient.setup(false);
+		DiseaseType diseaseType = testDiseaseType.setup(false);
+
+		Disease disease = testDisease.setup(diseaseType, false);
+		disease.setCode("angal.opd.alldisease");
+		disease.getType().setCode("angal.opd.alltype");
+
+		Opd opd = testOpd.setup(patient, disease, true);
+		GregorianCalendar now = new GregorianCalendar();
+		opd.setVisitDate(now);
+
+		patientIoOperationRepository.saveAndFlush(patient);
+		diseaseTypeIoOperationRepository.saveAndFlush(diseaseType);
+		diseaseIoOperationRepository.saveAndFlush(disease);
+		opdIoOperationRepository.saveAndFlush(opd);
+
+		ArrayList<Opd> opds = opdBrowserManager.getOpdList(0);
+		assertThat(opds.get(opds.size() - 1).getCode()).isEqualTo(opd.getCode());
+	}
+
+	@Test
+	public void testMgrGetOpdToday() throws Exception {
+		Patient patient = testPatient.setup(false);
+		DiseaseType diseaseType = testDiseaseType.setup(false);
+
+		Disease disease = testDisease.setup(diseaseType, false);
+		disease.setCode("angal.opd.alldisease");
+		disease.getType().setCode("angal.opd.alltype");
+
+		Opd opd = testOpd.setup(patient, disease, true);
+		GregorianCalendar now = new GregorianCalendar();
+		opd.setVisitDate(now);
+
+		patientIoOperationRepository.saveAndFlush(patient);
+		diseaseTypeIoOperationRepository.saveAndFlush(diseaseType);
+		diseaseIoOperationRepository.saveAndFlush(disease);
+		opdIoOperationRepository.saveAndFlush(opd);
+
+		ArrayList<Opd> opds = opdBrowserManager.getOpd(false);
+		assertThat(opds.get(opds.size() - 1).getCode()).isEqualTo(opd.getCode());
```

Comment:
Also here we could add cases out of the range

Reply:
done; and I added additional out of range objects to two more tests not noted here.

Reply:
thanks a lot!